### PR TITLE
Add script to sync kubefed CRD manifests

### PR DESCRIPTION
**Pull request by font**
This adds the script to generate the kubefed CRD manifests and then synchronize them with the ones stored in this repo. I've removed the old kubefed CRD file names as I don't think their names were part of any naming convention and have added the newly generated ones with the more conventional naming of `<group>_<version>_<kind>_crd.yaml`.

### CODE REVIEW

File: scripts/sync-manifests.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+export GOPATH="${TMP_GENDIR}"
+go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd
+
+# Rename CRDs to convention used by CRD generation code:
+# <group>_<version>_<kind>_crd.yaml
+for file in config/crds/*.yaml; do
+  mv "${file}" "${file%.yaml}_crd.yaml"
+done
+
+# Strip off 'v' prefix and anything after the last '-' e.g. v0.1.0-rc4 -> 0.1.0
```
Comment (pmorie) [inline scripts/sync-manifests.sh:49]: @abhat @sohankunkerkar do we want to drop 0.1.0-rcX -> 0.1.0 here?

File: scripts/sync-manifests.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+kubefed_version_crd_dir="${KUBEFED_VERSION#v}"
+kubefed_version_crd_dir="${kubefed_version_crd_dir%-*}"
+
+# Copy CRDs to respective directories
+crd_dirs="${ROOT_DIR}/deploy/crds"
+crd_dirs+=" ${ROOT_DIR}/deploy/olm-catalog/kubefed-operator/${kubefed_version_crd_dir}"
+for dir in ${crd_dirs}; do
+  if [[ ! -d "${dir}" ]]; then
```
Comment (pmorie) [inline scripts/sync-manifests.sh:57]: How about we just do an mkdir -P here instead?

**Review by pmorie — COMMENTED** [review]
Very minor comment about the update script, as well as one important question

File: scripts/sync-manifests.sh (hunk 2 above)
Comment (font) [reply scripts/sync-manifests.sh:57]: Yeah it was a tossup and decided against it for reasons. I'll make the change.

Comment (font) [reply scripts/sync-manifests.sh:57]: Updated to echo a message that the dir doesn't exist and then create it.

File: scripts/sync-manifests.sh (hunk 1 above)
Comment (abhat) [reply scripts/sync-manifests.sh:49]: Right now we aren't maintaining a per RC version of the CRDs. So I think this is needed.

**Pull request merged**